Assets/Scripts/GameState/LeaderboardData.cs

```diff
@@ -56,58 +75,54 @@ public static void SaveScores(Highscores scores)
         stream.Close();
     }
 
-    public static Highscores LoadScores()
+    /// <summary>
+    /// Load all saved scores
+    /// </summary>
+    /// <returns>Returns a ordered list of Highscore structs</returns>
+    public static List<Highscore> LoadScores()
     {
         if (File.Exists(path))
         {
             BinaryFormatter formatter = new BinaryFormatter();
             FileStream stream = new FileStream(path, FileMode.Open);
 
-            Highscores data = formatter.Deserialize(stream) as Highscores;
-            stream.Close();
+            List<Highscore> data;
 
-            if(data.name.Length < 10 || data.score.Length < 10)
+            // Check if the file is corrupted
+            try
             {
-                // Scores are missing, delete the data and reset
+                data = formatter.Deserialize(stream) as List<Highscore>;
+            }
+            catch
+            {
+                stream.Close();
                 File.Delete(path);
-                data = LoadScores();
+                Debug.LogWarning("Highscore data was corrupted, it has been replaced.");
+                return LoadScores();
             }
 
+            stream.Close();
```

Comment:
This `Close()` statement (and the one inside of the `catch` block above) should be replaced by one inside of a `finally` block:
```suggestion
            } finally
            {
                stream.Close();
            }
```
(This code will be run even if the function returns inside of the `try` or `catch` block, in case you didn't know 😊)